invenio_app_rdm/requests_ui/views/requests.py

```diff
@@ -30,18 +30,31 @@
 from invenio_app_rdm.records_ui.views.deposits import load_custom_fields
 
 
-def _resolve_topic_draft(request):
-    """Resolve the record in the topic when it is a draft."""
+def _resolve_topic_record(request):
+    """Resolve the record in the topic, when it is a draft or a published record."""
     user_owns_request = str(request["expanded"]["created_by"]["id"]) == str(
         current_user.id
     )
-
     if request["is_closed"] and not user_owns_request:
         return dict(permissions={}, record_ui=None)
 
-    recid = ResolverRegistry.resolve_entity_proxy(request["topic"])._parse_ref_dict_id()
+    record = None
+    # parse the topic field to get the draft/record pid `record:abcd-efgh`
+    entity = ResolverRegistry.resolve_entity_proxy(request["topic"])
+    pid = entity._parse_ref_dict_id()
     try:
-        record = current_rdm_records_service.read_draft(g.identity, recid, expand=True)
+        # read draft
+        record = current_rdm_records_service.read_draft(g.identity, pid, expand=True)
+    except NoResultFound:
+        try:
+            # read published record
+            record = current_rdm_records_service.read(g.identity, pid, expand=True)
```

Comment:
I think we added `.exists`to the service at some point, maybe it would simplify these blocks

Reply:
we actually need the returned `record`: using the `exists` will then require to 2 queries :(